### PR TITLE
feat: update to oclif v3

### PR DESCRIPTION
**Pull request by RodEsp**
> This needs to be merged at the same time as https://github.com/salesforcecli/github-workflows/pull/10

### What does this PR do?
Updates oclif to v3

In order to do this without any breaking changes I added a GHA that renames the `.pkg` produced by oclif v3 to have the same name as the `.pkg` produced by oclif v2.

Example of Successful GHA Runs releasing to `dev-rc` and `dev` channels for testing:
* [RC release](https://github.com/salesforcecli/sfdx-cli/actions/runs/3412333251)
* [Promote RC to latest](https://github.com/salesforcecli/sfdx-cli/actions/runs/3412737032)
> The failed jobs in those workflows are irrelevant for the testing.

### What issues does this PR fix or reference?
@W-10989572@

### CODE REVIEW

File: .github/workflows/promote.yml [hunk 1]
```diff
@@ -62,17 +62,33 @@ jobs:
           AWS_ACCESS_KEY_ID: ${{secrets.AWS_ACCESS_KEY_ID}}
           AWS_SECRET_ACCESS_KEY: ${{secrets.AWS_SECRET_ACCESS_KEY}}
 
+  # This job is renames the sfdx-x64.pkg that oclif v3 produces to sfdx.pkg which matches what oclif v2 produces.
+  # Doing so avoids breaking changes for customers relying on the old URLs to download the CLI.
+  # It can be removed once the developer site is linking to the new sfdx-x64.pkg file and the mac signing job is signing the new -x64.pkg file.
+  rename-mac-pkg:
+    runs-on: ubuntu-latest
+    env:
```
Comment (mshanemc) [inline .github/workflows/promote.yml:70]: how does this machine get aws?  or is it built into the ubuntu-latest?
https://github.com/salesforcecli/github-workflows/pull/10/files# doesn't do anything to make that happen.

the pack/upload mac job gets it via checkout, node, then yarn install (the CLI has it as a devDep)
https://github.com/salesforcecli/github-workflows/blob/60225d372a5ddca0fa936a6984e0d91057964bab/.github/workflows/packUploadMac.yml#L30

Comment (RodEsp) [reply .github/workflows/promote.yml:70]: Yeah, it's built into `ubuntu-latest`.

Comment (RodEsp) [reply .github/workflows/promote.yml:70]: https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#cli-tools

File: .github/workflows/rcBuild.yml [hunk 2]
```diff
@@ -54,7 +69,7 @@ jobs:
 
   Post-RC-to-Slack:
     runs-on: ubuntu-latest
-    needs: [pack-verify-upload-tarballs, npm-release, pack-upload-win, pack-upload-mac]
+    needs: [pack-verify-upload-tarballs, npm-release, pack-upload-win, pack-upload-mac, rename-mac-pkg]
```
Comment (mshanemc) [inline .github/workflows/rcBuild.yml:72]: you probably don't need pack-upload-mac then, just the rename?

**Pull request merged**